src/banded.jl

```diff
@@ -0,0 +1,550 @@
+
+    return reshape(xa, Nqh, n, Neh)
+end
+
+"""
+    batchedbandedlu!(A::AbstractArray{T, 4} [, kl = div(size(mat, 2) - 1, 2)])
+
+Compute the banded LU factors of the batched matrix `A`. Uses the `A` as storage
+for the factors. The optional argument `kl` is the lower bandwidth of the
+matrix; upper bandwidth is calculated as `ku = size(mat, 2) - 1 - kl`.
+
+The matrix `A` should have indexing `A[i, b, v, h]` where the indices `i` and
+`h` are the batch indices with the size of the first index defining the kernel 
+workgroup size the last index the number of workgroups. The index `b` is the
+band index for column `v`; storage uses the [LAPACK
+format](https://www.netlib.org/lapack/lug/node124.html).
+
+For example the matrix `A` would be stored in banded storage `B` with lower
+bandwidth `kl = 2`.
+
+```reply
+julia> A = [11 12  0  0  0  0  0
+            21 22 23  0  0  0  0
+            31 32 33 34  0  0  0
+            41 42 43 44 45  0  0
+             0 52 53 54 55 56  0
+             0  0 63 64 65 66 67
+             0  0  0 74 75 76 77]
+7×7 Matrix{Int64}:
+ 11  12   0   0   0   0   0
+ 21  22  23   0   0   0   0
+ 31  32  33  34   0   0   0
+ 41  42  43  44  45   0   0
+  0  52  53  54  55  56   0
+  0   0  63  64  65  66  67
+  0   0   0  74  75  76  77
+
+julia> B = [ 0 12 23 34 45 56 66 67
+            11 22 33 44 55 65 77  0
+            21 32 43 54 64 76  0  0
+            31 42 53 63 75  0  0  0
+            41 52  0 74  0  0  0  0]
+5×8 Matrix{Int64}:
+  0  12  23  34  45  56  66  67
+ 11  22  33  44  55  65  77   0
+ 21  32  43  54  64  76   0   0
+ 31  42  53  63  75   0   0   0
+ 41  52   0  74   0   0   0   0
+```
+"""
+function batchedbandedlu!(
+        mat::A,
+        kl = div(size(mat, 2) - 1, 2)
+    ) where {T, A <: AbstractArray{T, 4}}
+    (Nqh, width, n, Neh) = size(mat)
+    ku = width - 1 - kl
+    fac = BatchedBandedLU{Nqh, n, ku, kl, Neh, T, A}(mat)
+    return batchedbandedlu!(fac)
+end
+
+function batchedbandedlu!(
+        fac::BatchedBandedLU{Nqh, n, ku, kl, Neh, T, A}
+    ) where {Nqh, n, ku, kl, Neh, T, A}
+    event = Event(device(A))
+    kernel! = bandedlu_kernel!(device(A), (Nqh,))
+    event = kernel!(fac; ndrange = (Nqh * Neh,), dependencies = (event,))
+    wait(event)
+    return fac
+end
+
+@kernel function bandedlu_kernel!(
+        fac::BatchedBandedLU{Nqh, n, ku, kl, Neh}
+    ) where{Nqh, n, ku, kl, Neh}
+
+    @uniform begin
+        # center index of band
+        c = ku + 1
+    end
+
+    # horizonal element number
+    eh = @index(Group, Linear)
+
+    # horizontal degree of freedom
+    ij = @index(Local, Linear)
+
+    # Create an object that indexes like a matrix for this thread
+    U = L = view(fac, ij, :, :, eh)
+
+    # matrix index: (u,v) -> banded index: (c + u - v, u)
+    # v is column index
+    @inbounds for v = 1:n
+        # Get the pivot
+        # Uvv = U[c, v]
+        Uvv = U[v, v]
+
+        # Fill L
+        for p = 1:kl
```

Comment:
```suggestion
        @unroll for p = 1:kl
```
Not sure if this is faster or not.